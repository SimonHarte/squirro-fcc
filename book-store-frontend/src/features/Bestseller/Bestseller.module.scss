@import "../../styles/variables.scss";
@import "../../styles/helpers.scss";

.wrapper {
  margin: ($spacing-base * 3) (-$spacing-base * 3) 0;
  padding: $spacing-base * 3;
  border-top: 1px solid lightgray;
  border-bottom: 1px solid lightgray;

  @include responsive {
    margin: ($spacing-base * 5) 0 0;
    padding: 0;
    border: 0;
  }
}

.table {
  width: 100%;
}

.caption {
  margin-bottom: $spacing-base * 2;
  font-style: italic;
}

.book {
  font-weight: bold;
}

.author {
  text-align: right;
}

.row + .row {
  @include responsive {
    border-top: 1px solid lightgray;
  }
}

.cell {
  padding: $spacing-base 0;
}
