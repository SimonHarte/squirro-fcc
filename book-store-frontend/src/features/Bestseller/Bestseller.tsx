import { type ReactElement, type FC } from 'react';
import classNames from 'classnames';
import { useGetBestsellers } from '../../app/hooks';
import type { BestSeller } from '../../types/Books';
import type { BestsellerProps } from './Bestseller.types';
import styles from './Bestseller.module.scss';

const Bestseller: FC<BestsellerProps> = ({
  books,
}: BestsellerProps): ReactElement => {
  const bestsellers = useGetBestsellers(books);

  return (
    <div className={styles.wrapper}>
      <table className={styles.table}>
        <caption className={styles.caption}>Best-selling books</caption>
        <tbody>
          {bestsellers.length > 0 ? (
            bestsellers.map(({ name, author }: BestSeller) => (
              <tr key={name} className={styles.row}>
                <td className={classNames(styles.cell, styles.book)}>{name}</td>
                <td className={classNames(styles.cell, styles.author)}>
                  {author}
                </td>
              </tr>
            ))
          ) : (
            <tr className={styles.row}>
              <td className={styles.cell}>No data available</td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  );
};

export default Bestseller;
