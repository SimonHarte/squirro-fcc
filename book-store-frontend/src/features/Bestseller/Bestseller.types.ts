import { BestSeller } from '../../types/Books';

export type BestsellerProps = {
  books: Array<BestSeller>;
};
