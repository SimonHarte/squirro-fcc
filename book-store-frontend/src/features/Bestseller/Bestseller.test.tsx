import { shallow } from 'enzyme';
import Bestseller from './index';

describe('Bestseller', () => {
  test('should render correctly', () => {
    const wrapper = shallow(
      <Bestseller
        books={[
          { name: 'Book 1', author: 'Author 1' },
          { name: 'Book 2', author: 'Author 2' },
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly without data', () => {
    const wrapper = shallow(<Bestseller books={[]} />);

    expect(wrapper).toMatchSnapshot();
  });
});
