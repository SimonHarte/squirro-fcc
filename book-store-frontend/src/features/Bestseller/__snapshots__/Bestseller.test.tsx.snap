// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bestseller should render correctly 1`] = `
<div
  className="wrapper"
>
  <table
    className="table"
  >
    <caption
      className="caption"
    >
      Best-selling books
    </caption>
    <tbody>
      <tr
        className="row"
        key="Book 1"
      >
        <td
          className="cell book"
        >
          Book 1
        </td>
        <td
          className="cell author"
        >
          Author 1
        </td>
      </tr>
      <tr
        className="row"
        key="Book 2"
      >
        <td
          className="cell book"
        >
          Book 2
        </td>
        <td
          className="cell author"
        >
          Author 2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Bestseller should render correctly without data 1`] = `
<div
  className="wrapper"
>
  <table
    className="table"
  >
    <caption
      className="caption"
    >
      Best-selling books
    </caption>
    <tbody>
      <tr
        className="row"
      >
        <td
          className="cell"
        >
          No data available
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
